recipes-omnect/initrdscripts/omnect-os-initramfs/flash-mode-2

```diff
@@ -122,32 +119,32 @@ run_bmap_tool() {
 
 # deploy WIC image via network to (same) active block device
 run_flash_mode_2() {
-    local root_a_p=""
-    local machine_features=""
     msg "Entering omnect flashing mode 2 ..."
 
     # disable flash mode for subsequent restarts
     set_bootloader_env_var ${OMNECT_FLASH_MODE_VAR}
 
-    if [ "${OMNECT_FLASH_MODE_ETH}" == "UNDEFINED" ]; then
-        msg_fatal "missing definition OMNECT_FLASH_MODE_ETH"
-        return 1
+    # we already mounted /dev/omnect/boot when we're using grub. so we unmount before copying boot.
+    if [ -f /usr/bin/grub-editenv ]; then
+        sync
+        run_cmd umount ${ROOTFS_DIR}/boot
     fi
 
     setup_fifo || return 1
-    setup_network ${OMNECT_FLASH_MODE_ETH} || return 1
+    setup_network eth0 || return 1
     run_bmap_tool || return 1
 
-    machine_features=$(grep ^MACHINE_FEATURES= /etc/os-release | awk -F'"' '{print $2}')
-    if [[ " ${machine_features} " =~ " grub " ]]; then
-        run_cmd mkdir -p /tmp/boot
-        run_cmd mount /dev/omnect/boot /tmp/boot
-        root_a_p=$(readlink -f /dev/omnect/rootA)
-        root_a_p=${root_a_p::-1}
-        run_cmd sed "s#@@ROOT_DEVICE@@#${root_a_p}#g" /etc/omnect/grub.cfg.in > /tmp/boot/EFI/BOOT/grub.cfg
-        run_cmd umount /tmp/boot
+    local dest_blk=$(readlink -f /dev/omnect/rootA)
+    dest_blk=${dest_blk::-1} # can be /dev/nvme0n1p or /dev/sda
+    local p=${dest_blk:0-1}
+    if [ "${p}" == "p" ]; then
+        dest_blk=${dest_blk::-1}
+    else
+        p=""
```

Comment:
just to clarify: only needed for efi_grub handling?

Reply:
yes its only needed for efi+grub handling.

but imho can not be moved. in flash-mode-1 these params are already known.
this was my idea to reuse as much code as i can regarding the efi+grub handling in both flash-modes.

perhaps a better way would be to have dedicated flash-mode implementations for u-boot vs efi+grub devices, but this imho worth a dedicated pbi. if we use dedicated implementations, we should do this for all scripts.
since the handling of reading/writing grub env vars is complicated, we would have to split most script files into multiple files (common vs special parts).  currently this is not in focus for this pbi.

Reply:
ok, let's use it at the moment as it is.